google-cloud-video-transcoder/samples/create_job_with_animated_overlay.rb

```diff
@@ -0,0 +1,148 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def create_job_with_animated_overlay project_id:, location:, input_uri:, overlay_image_uri:, output_uri:
+  # [START transcoder_create_job_with_animated_overlay]
+  # project_id  = "YOUR-GOOGLE-CLOUD-PROJECT"  # (e.g. "my-project")
+  # location    = "YOUR-JOB-LOCATION"  # (e.g. "us-central1")
+  # input_uri   = "YOUR-GCS-INPUT-VIDEO"  # (e.g. "gs://my-bucket/my-video-file")
+  # overlay_image_uri   = "YOUR-GCS-OVERLAY-IMAGE"  # (e.g. "gs://my-bucket/overlay.jpg")
+  # output_uri  = "YOUR-GCS-OUTPUT-FOLDER/"  # (e.g. "gs://my-bucket/my-output-folder/")
+
+  # Require the Transcoder client library.
+  require "google/cloud/video/transcoder"
+
+  # Create a Transcoder client.
+  client = Google::Cloud::Video::Transcoder.transcoder_service
+
+  # Build the resource name of the parent.
+  parent = client.location_path project: project_id, location: location
+
+  # Build the job config.
+  new_job = {
+    config: {
+      elementary_streams: [
+        {
+          key: "video-stream0",
+          video_stream: {
+            h264: {
+              height_pixels: 360,
+              width_pixels: 640,
+              bitrate_bps: 550_000,
+              frame_rate: 60
+            }
+          }
+        },
+        {
+          key: "audio-stream0",
+          audio_stream: {
+            codec: "aac",
+            bitrate_bps: 64_000
+          }
+        }
+      ],
+      mux_streams: [
+        {
+          key: "sd",
+          container: "mp4",
+          elementary_streams: [
+            "video-stream0",
+            "audio-stream0"
+          ]
+        }
+      ],
+      overlays: [
+        {
+          image: {
+            uri: overlay_image_uri,
+            resolution: {
+              x: 0,
+              y: 0
+            },
+            alpha: 1
+          },
+          animations: [
+            {
+              animation_fade: {
+                fade_type: Google::Cloud::Video::Transcoder::V1::Overlay::FadeType::FADE_IN,
+                xy: {
+                  x: 0.5,
+                  y: 0.5
+                },
+                start_time_offset: {
+                  seconds: 5
+                },
+                end_time_offset: {
+                  seconds: 10
+                }
+              }
+            },
+            {
+              animation_fade: {
+                fade_type: Google::Cloud::Video::Transcoder::V1::Overlay::FadeType::FADE_OUT,
+                xy: {
+                  x: 0.5,
+                  y: 0.5
+                },
+                start_time_offset: {
+                  seconds: 12
+                },
+                end_time_offset: {
+                  seconds: 15
+                }
+              }
+            }
+          ]
+        }
+      ]
+    }
+  }
+  new_job[:input_uri] = input_uri
+  new_job[:output_uri] = output_uri
```

Comment:
Why not just include these in the initializer above? (Same for a bunch of the following samples.)

Reply:
Updated samples that use these fields